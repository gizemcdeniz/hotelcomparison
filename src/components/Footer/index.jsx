import React from "react";
import "./index.scss";
import { Container, Row, Col } from "react-bootstrap";
import { useTranslation } from "react-i18next";
import logo from "../../images/logo1.png";
import twitterIcon from "../../images/icons/twitterIcon.svg";
import linkedInIcon from "../../images/icons/linkedInIcon.svg";
import instagramIcon from "../../images/icons/instagramIcon.svg";
import facebookIcon from "../../images/icons/facebookIcon.svg";

const Footer = () => {
  const { t } = useTranslation();

  return (
    <footer className="mainFooter">
      <Container>
        <Row>
          <Col>
            <img src={logo} alt="Logo" className="logo" />
          </Col>
          <Col md={3} className="footerInfo">
            <h5>{t("footer.followUs")}</h5>
            <Row>
              <a href="https://www.twitter.com/">
                <img src={twitterIcon} alt="Twitter icon" />
              </a>
              <a href="https://www.linkedin.com/">
                <img src={linkedInIcon} alt="LinkedIn icon" />
              </a>
              <a href="https://www.instagram.com/">
                <img src={instagramIcon} alt="Instagram icon" />
              </a>
              <a href="https://www.facebook.com/">
                <img src={facebookIcon} alt="Facebook icon" />
              </a>
            </Row>
          </Col>
        </Row>
      </Container>
    </footer>
  );
};

export default Footer;
