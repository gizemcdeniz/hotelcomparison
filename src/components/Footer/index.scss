
@import "../../style/variables";
@import "../../style/bootstrapImports";

.mainFooter {
  background-color: #1D3557;
  padding: 2rem 0;
  font-family: $mainFontFamily;

  .row {
    align-items: center;
    justify-content: flex-start;
  }

  .logo {
    width: 200px;
  }

  .footerInfo {
    color: $lightColor;
    display: flex;
    flex-direction: column;
    font-weight: 500;
    align-items: flex-end;
    padding: 0rem;

    a:hover {
      color: $primaryColor;
      cursor: pointer;
    }

    img {
      width: 1.5rem;
      margin: 0 1.3rem;
    }

    img:hover {
      background-color: $primaryColor;
      padding: 0.1rem;
      border-radius: 0.2rem;
    }
  }

  a {
    color: $lightColor;
  }
}

  .logo-text {
    color: #fff;
  }

@include media-breakpoint-down(sm) {
  .mainFooter {
    .row {
      text-align: center;
    }

    .footerInfo {
      align-items: center;
      margin-top: 1rem;
    }
  }
}
