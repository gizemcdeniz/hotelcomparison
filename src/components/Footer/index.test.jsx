import React from "react";
import renderer from "react-test-renderer";
import Footer from "./index";

it("renders correctly and matches the snapshot", () => {
  const tree = renderer.create(<Footer />).toJSON();
  expect(tree).toMatchSnapshot();
});
