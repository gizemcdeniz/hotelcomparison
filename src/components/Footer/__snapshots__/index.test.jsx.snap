// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and matches the snapshot 1`] = `
<footer
  className="mainFooter"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <img
          alt="Logo"
          className="logo"
          src="logoFooter.png"
        />
      </div>
      <div
        className="footerInfo col-md-3"
      >
        <h5>
          footer.followUs
        </h5>
        <div
          className="row"
        >
          <a
            href="https://www.twitter.com/"
          >
            <img
              alt="Twitter icon"
              src="twitterIcon.svg"
            />
          </a>
          <a
            href="https://www.linkedin.com/"
          >
            <img
              alt="LinkedIn icon"
              src="linkedInIcon.svg"
            />
          </a>
          <a
            href="https://www.instagram.com/"
          >
            <img
              alt="Instagram icon"
              src="instagramIcon.svg"
            />
          </a>
          <a
            href="https://www.facebook.com/"
          >
            <img
              alt="Facebook icon"
              src="facebookIcon.svg"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</footer>
`;
